tests/app/shared/issue-tables/search-filter.spec.ts

```diff
@@ -0,0 +1,89 @@
+        const dummyTeam: Team = new Team({
+        id: 'dummyId',
+        teamMembers: [],
+        });
+        let dummySearchKey: string;
+        const mediumSeverityIssueWithResponse: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+        mediumSeverityIssueWithResponse.responseTag = 'Accepted';
+        const mediumSeverityIssueWithAssigneee: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_ASSIGNEES, dummyTeam);
+        const lowSeverityFeatureFlawIssue: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY, dummyTeam);
+        const highSeverityDocumentationBugIssue: Issue =
+            Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+        const duplicatedIssue: Issue = Issue.createPhaseTeamResponseIssue(DUPLICATED_ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+        duplicatedIssue.duplicateOf = highSeverityDocumentationBugIssue.id;
+
+        const issuesList: Issue[] = [
+            mediumSeverityIssueWithResponse,
+            mediumSeverityIssueWithAssigneee,
+            lowSeverityFeatureFlawIssue,
+            highSeverityDocumentationBugIssue
+        ];
+        const displayedColumn: string[] = [
+            TABLE_COLUMNS.ID,
+            TABLE_COLUMNS.TITLE,
+            TABLE_COLUMNS.TYPE,
+            TABLE_COLUMNS.SEVERITY,
+            TABLE_COLUMNS.RESPONSE,
+            TABLE_COLUMNS.ASSIGNEE,
+            TABLE_COLUMNS.DUPLICATED_ISSUES,
+        ];
+        const issueService: IssueService = new IssueService(null, null, null, null, null, null);
+
+        beforeEach(() => {
+            issueService.updateLocalStore(mediumSeverityIssueWithResponse);
+            issueService.updateLocalStore(mediumSeverityIssueWithAssigneee);
+            issueService.updateLocalStore(lowSeverityFeatureFlawIssue);
+            issueService.updateLocalStore(highSeverityDocumentationBugIssue);
+            issueService.updateLocalStore(duplicatedIssue);
+        });
+
+        it('returns filtered list of issues which includes issues that contain the search key in any of its assignees', () => {
+            dummySearchKey = 'anubh-v';
+            expect(applySearchFilter(dummySearchKey, displayedColumn, issueService, issuesList))
+                .toEqual([mediumSeverityIssueWithAssigneee]);
+        });
+
+        it('returns filtered list of issues which includes issues that contain the search key in any of its duplicate issues id', () => {
+            dummySearchKey = duplicatedIssue.id.toString();
+            expect(applySearchFilter(dummySearchKey, displayedColumn, issueService, issuesList))
+                .toEqual([highSeverityDocumentationBugIssue]);
+        });
+
+        it('returns filtered list of issues which includes issues that contain the search key in any other column', () => {
+            // Checks id of issue
+            dummySearchKey = '92';
```

Comment:
Perhaps instead of hard-coding the value we could reference it from the `mediumSeve..Response` object? So like `mediumSeverityIssueWithAssignees.id.toString()`

Reply:
Yup, I should have done that😅